x-pack/platform/test/alerting_api_integration/spaces_only/tests/alerting/group3/builtin_alert_types/es_query/rule.ts

```diff
@@ -183,80 +182,124 @@ export default function ruleTests({ getService }: FtrProviderContext) {
       [
         'esQuery',
         async () => {
-          await createRule({
-            name: 'never fire',
-            esQuery: `{\n  \"query\":{\n    \"match_all\" : {}\n  }\n}`,
-            size: 100,
-            thresholdComparator: '<',
-            threshold: [0],
-            aggType: 'avg',
-            aggField: 'testedValue',
-          });
-          await createRule({
+          return await createDSLRule(supertest, objectRemover, connectorId, {
             name: 'always fire',
-            esQuery: `{\n  \"query\":{\n    \"match_all\" : {}\n  }\n}`,
-            size: 100,
             thresholdComparator: '>',
             threshold: [-1],
-            aggType: 'avg',
-            aggField: 'testedValue',
           });
         },
       ] as const,
       [
         'searchSource',
         async () => {
-          const esTestDataView = await indexPatterns.create(
-            { title: ES_TEST_INDEX_NAME, timeFieldName: 'date' },
-            { override: true },
-            getUrlPrefix(Spaces.space1.id)
-          );
-          await createRule({
-            name: 'never fire',
-            size: 100,
-            thresholdComparator: '<',
-            threshold: [0],
-            searchType: 'searchSource',
-            searchConfiguration: {
-              query: {
-                query: '',
-                language: 'kuery',
-              },
-              index: esTestDataView.id,
-              filter: [],
+          const esTestDataViewId = await createIndexPattern('date');
+          return await createSearchSourceRule(
+            supertest,
+            objectRemover,
+            connectorId,
+            {
+              name: 'always fire',
+              thresholdComparator: '>',
+              threshold: [-1],
             },
-            aggType: 'avg',
-            aggField: 'testedValue',
-          });
-          await createRule({
+            esTestDataViewId
+          );
+        },
+      ] as const,
+    ].forEach(([searchType, initData]) =>
+      it(`runs correctly: alerts as expected for 0 documents for ${searchType} search type`, async () => {
```

Comment:
created specific test for testing that 0 documents will matching the `> -1` condition and updated all other test conditions to be `> 0`